server/auth/middleware.js

```diff
@@ -17,6 +19,9 @@ const isAuthenticated = (req, res, next) => {
 
 const isSuperUser = (req, res, next) => {
   if (req.user?.isSuperUser) {
+    if (!req.session.twoFactor) {
+      res.status(HttpStatus.UNAUTHORIZED).end();
```

Comment:
You need to restructure the conditional or add a return statement, otherwise, it will send UNAUTHORIZED and fall through and call next().

Reply:
Looks like inserting a simple else before next()  would solve the issue here. Thanks for the keen eye here!